middleware/src/ipcnotification/ipcnotification.go

```diff
@@ -0,0 +1,131 @@
+// Package ipcnotification reads notifications from a named pipe and passes them
+// into a channel.
+package ipcnotification
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"syscall"
+)
+
+// Notification represents an IPC notification that is passed via a named pipe.
+type Notification struct {
+	Version int         `json:"version"`
+	Topic   string      `json:"topic"`
+	Payload interface{} `json:"payload"`
+}
+
+func (notification *Notification) String() string {
+	return fmt.Sprintf("[Version: %d, Topic: %s, Payload: %v]", notification.Version, notification.Topic, notification.Payload)
+}
+
+// Reader reads IPCNotifications from a named pipe into a channel.
+type Reader struct {
+	Notifications chan Notification
+	closing       bool
+	filePath      string
+	namedPipe     *os.File
+}
+
+// NewReader returns a new Reader that starts reading from the named pipe passed.
+func NewReader(filepath string) (*Reader, error) {
+	reader := &Reader{
+		Notifications: make(chan Notification),
```

Comment:
I would make this private and expose it through a getter, but that is just my personal taste.

Reply:
I agree with making it private. No need to be potentially reassigned by somebody else.